doc/ACL/SAI-Proposal-ACL-Bincode.md

```diff
@@ -0,0 +1,159 @@
+#  ACL BINCODE Field Matching
+-------------------------------------------------------------------------------
+ Title       | ACL BINCODE Field Matching
+-------------|-----------------------------------------------------------------
+ Authors     | Nader Shinouda, Cisco
+ Status      | In review
+ Type        | Standards track
+ Created     | 2024-07-01 - Initial Draft
+ SAI-Version | 1.14
+-------------------------------------------------------------------------------
+
+
+## 1.0  Introduction ##
+
+This spec enhances the existing ACL spec to add support for bincode field matching. Bincode is part of prefix compression, where an IP prefix is mapped to Bincode.
+
+New table attributes allow setting a source and destination prefix compression tables on the creation of an ACL table. New field entry attributes allow for matching on a specific bincode from either the source or destionation prefix tables.
+
+## 2.0 Specification ##
+
+New table attributes allow for setting both the source and destination prefix compression tables that will be used in field matching
+```c
+    /**
+     * @brief SRC BINCODE
+     *
+     * This key is dedicated to matching on a SRC BINCODE
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_SRC_PREFIX_BINCODE
+
+    /**
+     * @brief DST BINCODE
+     *
+     * This key is dedicated to matching on a DST BINCODE
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_DST_PREFIX_BINCODE
+
+    /**
+     * @brief SRC prefix Table Object ID
+     *
+     * An object pointer to a prefix table used for
+     * source prefix lookups
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_PREFIX_COMPRESSION_TABLE
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_ACL_TABLE_ATTR_SRC_PREFIX_COMPRESSION_TABLE,
+
+    /**
+     * @brief DST prefix Table Object ID
+     *
+     * An object pointer to a prefix table used for
+     * destination prefix lookups
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_PREFIX_COMPRESSION_TABLE
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_ACL_TABLE_ATTR_DST_PREFIX_COMPRESSION_TABLE,
+```
+
+New field entry attributes allow for lookups based on a bincode value.
+
+```c
+   /**
+     * @brief SRC BINCODE
+     *
+     * @type sai_acl_field_data_t sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default disabled
+     */
+    SAI_ACL_ENTRY_ATTR_FIELD_SRC_PREFIX_BINCODE,
+
+    /**
+     * @brief DST BINCODE
+     *
+     * @type sai_acl_field_data_t sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default disabled
+     */
+    SAI_ACL_ENTRY_ATTR_FIELD_DST_PREFIX_BINCODE,
```

Comment:
ACL qualification in your case can be either or DST_PREFIX_BINCODE or SRC_PREFIX_BINCODE  or can both be qualified simultaneously ? If former is true a generic name like META_DATA should suffice.

Reply:
@rck-innovium, I updated the documentation. I combined both in a single document. Please let me know if this meets your request and resolve if it does